source/grid.js

```diff
@@ -18,22 +18,24 @@
 
 this.initialize = function() {
     //Set up the grid map as a 2D array
-    this.tiles = [];
-    for ( var i = 0; i < this.maxX - this.minX; i++ ) {
-        this.tiles[ i ] = [];
-        for ( var j = 0; j < this.maxY - this.minY; j++ ) {
-            this.tiles[ i ][ j ] = new GridTile();
+    var tiles = this.tiles = new Array( this.maxX - this.minX );
+    for ( var i = 0; i < tiles.length; i++ ) {
+        tiles[ i ] = new Array( this.maxY - this.minY );
+        for ( var j = 0; j < tiles[ i ].length; j++ ) {
+            tiles[ i ][ j ] = new GridTile();
         }
     }
     this.future( 0 ).setBoundaryValues();
 }
 
 this.setBoundaryValues = function() {
-    if ( this.boundaryValues ) {
+    var boundaryValues = this.boundaryValues;
+    var tiles = this.tiles;
```

Comment:
Why copy these into local variables?  Is it just so you can avoid saying "this." in the rest of the code?  If so, I would suggest *not* doing it, because it is considerably less clear (the "this." signals to me that I'm looking at a member of the object, rather than a local variable).

Same comment throughout, obviously.